packages/ckeditor5-ui/src/toolbar/balloon/balloontoolbar.ts

```diff
@@ -275,6 +280,39 @@ export default class BalloonToolbar extends Plugin {
 		}
 	}
 
+	/**
+	 * Add or remove editable elements to the focus tracker. It watches added and removed roots
+	 * and adds or removes their editable elements to the focus tracker.
+	 */
+	private _trackEditablesFocusableElements() {
+		const { editor, focusTracker } = this;
+		const { ui } = editor;
+
+		// Add existing editable elements to the focus tracker.
+		for ( const editableName of ui.getEditableElementsNames() ) {
+			const editableElement = ui.getEditableElement( editableName );
+
+			focusTracker.add( editableElement! );
+		}
```

Comment:
I'd suggest avoiding such an approach if we bind to the `view.document.roots` collection early enough.